firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarMiddleware.swift

```diff
@@ -476,16 +490,11 @@ class ToolbarMiddleware: FeatureFlaggable {
 
     private func addressToolbarPageActions(
         action: ToolbarMiddlewareAction,
-        state: AppState,
+        toolbarState: borrowing ToolbarState,
```

Comment:
What's the reason for using `borrowing` here?

Reply:
It is an optimization technique that instructs the compiler to `borrow` the argument (in general the compiler copies it), avoiding the creation of a copy by taking ownership of the argument, but without the ability to modify it.